### DISCUSSION TITLE
Error Running LIS Executable - MPI_INIT Error

### DISCUSSION THREAD

**Discussion by Heejeong-Jeong** [NASA-LIS/LISF#1518] [Q&A]
Hello,

While the LIS compiling is successful, the following error occurs when trying to run the LIS executable with configuration files:

***The MPI_Comm_f2c() function was called before MPI_INIT was invoked.***
***This is disallowed by the MPI standard.***
***Your MPI job will now abort.***
[.] Local abort before MPI_INIT completed successfully, but I am not able to aggregate error messages, and I cannot guarantee that all other processes were killed.

LIS was configured with the following specifications, and related modules are installed as follows:

1. LIS Configuration:
[configure.lis.txt](https://github.com/NASA-LIS/LISF/files/14916212/configure.lis.txt)

2. Modules:
[module_list.txt](https://github.com/NASA-LIS/LISF/files/14916213/module_list.txt)
  
A similar issue was identified in #1392, and I attempted to resolve the error using the suggested methods but was unsuccessful. 
Is there a way to solve this problem? I would greatly appreciate any assistance you could provide.

**Comment by sudarsan3010** [comment]
Hi @Heejeong-Jeong,

Ensure that the ESMF and LIS are being compiled using same processors. Also, call and check 'which mpirun' to see whether the one expected is being called, deactivate anaconda environment if it is causing some problem. From your configuration file, I understood that ESMF has been compiled with mpiuni, which should be replaced with 'mpich', if you are compiling LIS with mpich.

Before installation of ESMF do the following,
for ESMF 8.1.1 it is - export ESMF_COMM=mpich3
for ESMF 8.5.0 it is - export ESMF_COMM=mpich 

To know your exact code check the following file,
esmf-8.5.0/build_config/Linux.gfortran.default/build_rules.mk

And, I have used the compiler flags as below,
export CC=mpicc
export CXX=mpicxx
export FC=mpif90
export F77=mpif77
